tests/jenkins/TestPackerBuild.groovy

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+import jenkins.tests.BuildPipelineTest
+import org.junit.Before
+import org.junit.Test
+import static com.lesfurets.jenkins.unit.MethodCall.callArgsToString
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.CoreMatchers.hasItem
+import static com.lesfurets.jenkins.unit.global.lib.GitSource.gitSource
+import static com.lesfurets.jenkins.unit.global.lib.LibraryConfiguration.library
+
+import static org.hamcrest.CoreMatchers.equalTo
+import static org.hamcrest.CoreMatchers.hasItems
+import static org.hamcrest.CoreMatchers.notNullValue
+
+class TestPackerBuild extends BuildPipelineTest {
+
+    // Variables
+    String packerBuildGitRespository = 'https://github.com/opensearch-project/opensearch-ci'
+    String packerBuildGitRespositoryReference = 'main'
+    String packerAgentJsonName = 'jenkins-agent-al2-arm64.json'
+
+    @Override
+    @Before
+    void setUp() {
+
+        helper.registerSharedLibrary(
+            library().name('jenkins')
+                .defaultVersion('2.2.0')
+                .allowOverride(true)
+                .implicit(true)
+                .targetPath('vars')
+                .retriever(gitSource('https://github.com/opensearch-project/opensearch-build-libraries.git'))
+                .build()
+            )
+
+        super.setUp()
+
+        binding.setVariable('PACKER_AGENT_JSON_NAME', packerAgentJsonName)
+        def sample_json = [
+            "variables" : [
+                "name-base" : "Jenkins-Agent-AL2-X64" ,
+                "os-version" : "AL2" ,
+                "build-region" : "us-east-1" ,
+                "build-vpc" : "vpc-<>" ,
+                "build-subnet" : "subnet-<>" ,
+                "build-secgrp" : "sg-<>" ,
+                "build-time" : "{{isotime \"2006-01-02T03-04-05Z\"}}" ,
```

Comment:
What does `build-time` mean? Asking if it needs to be dynamic while building the actual AMI?

Reply:
No need, it is a format to follow when print out in console.